service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/CreateAndDeleteAzureControlledResourceFlightTest.java

```diff
@@ -208,22 +193,63 @@ public void createAndDeleteAzureRelayNamespaceControlledResource() throws Interr
-            jobService.getStairway(),
-            CreateAzureContextFlight.class,
-            azureTestUtils.createAzureContextInputParameters(workspaceUuid, userRequest),
-            STAIRWAY_FLIGHT_TIMEOUT,
-            null);
+    final ApiAzureStorageCreationParameters creationParameters =
+            ControlledResourceFixtures.getAzureStorageCreationParameters();
 
-    assertEquals(FlightStatus.SUCCESS, createAzureContextFlightState.getFlightStatus());
-    assertTrue(
-        workspaceService.getAuthorizedAzureCloudContext(workspaceUuid, userRequest).isPresent());
+    final UUID resourceId = UUID.randomUUID();
+    ControlledAzureStorageResource resource =
+            ControlledAzureStorageResource.builder()
+                    .common(
+                            ControlledResourceFields.builder()
+                                    .workspaceUuid(workspaceUuid)
+                                    .resourceId(resourceId)
+                                    .name(getAzureName("rs"))
+                                    .description(getAzureName("rs-desc"))
+                                    .cloningInstructions(CloningInstructions.COPY_NOTHING)
+                                    .accessScope(AccessScopeType.fromApi(ApiAccessScope.SHARED_ACCESS))
+                                    .managedBy(ManagedByType.fromApi(ApiManagedBy.USER))
+                                    .build())
+                    .storageAccountName(creationParameters.getName())
+                    .region(creationParameters.getRegion())
+                    .build();
+
+    // Submit a storage account creation flight and then verify the resource exists in the workspace.
+    createResource(workspaceUuid, userRequest, resource, WsmResourceType.CONTROLLED_AZURE_STORAGE_ACCOUNT);
+
+    // Submit a storage account deletion flight.
+    FlightState deleteFlightState =
+            StairwayTestUtils.blockUntilFlightCompletes(
+                    jobService.getStairway(),
+                    DeleteControlledResourceFlight.class,
+                    azureTestUtils.deleteControlledResourceInputParameters(
+                            workspaceUuid, resourceId, userRequest, resource),
+                    STAIRWAY_FLIGHT_TIMEOUT,
+                    null);
+    assertEquals(FlightStatus.SUCCESS, deleteFlightState.getFlightStatus());
+
+    TimeUnit.SECONDS.sleep(5);
+    StorageManager manager = azureTestUtils.getStorageManager();
+    ManagementException exception =
+            assertThrows(
+                    ManagementException.class,
+                    () ->
+                            manager.storageAccounts().getByResourceGroup(
+                                    azureTestUtils.getAzureCloudContext().getAzureResourceGroupId(),
+                                    resource.getStorageAccountName()));
+    // We see both ResourceNotFound and NotFound in the code field
+    assertTrue(exception.getValue().getCode().contains("NotFound"));
```

Comment:
```exception.response.getStatusCode() == 404``` might be clearer (assuming both flavors of error alluded to in the comment on L244 are both HTTP 404s.